apm-agent-core/src/main/java/co/elastic/apm/agent/bci/IndyBootstrap.java

```diff
@@ -462,4 +489,12 @@ private static List<String> getClassNamesFromBundledPlugin(String adviceClassNam
         return pluginClasses;
     }
 
+    private static String getPluginPackageFromAdviceClass(String adviceClassName) {
+        if (!adviceClassName.startsWith(EMBEDDED_PLUGINS_PACKAGE_PREFIX)) {
+            throw new IllegalArgumentException("invalid advice class location : " + adviceClassName);
+        }
+        String pluginPackage = adviceClassName.substring(0, adviceClassName.indexOf('.', EMBEDDED_PLUGINS_PACKAGE_PREFIX.length()));
+        return pluginPackage;
+    }
```

Comment:
Why does this duplicate [what we have in the customizer](https://github.com/elastic/apm-agent-java/blob/9575d839280b71e6ef771be45e5436d241b68c04/apm-agent-core/src/main/java/co/elastic/apm/agent/bci/PluginClassLoaderRootPackageCustomizer.java#L35-L41)? Can't we reuse it?

Reply:
Well this code was simply extracted from the existing local `getClassNamesFromBundledPlugin` method. What would be the correct place for this logic in your opinion? A static method within `PluginClassLoaderRootPackageCustomizer`? Or somewhere else?

Reply:
Yes, this definitely seems suitable for a static method and `PluginClassLoaderRootPackageCustomizer` seems like a reasonable common place to compute the value that is both its base field AND is used as its mapping key. 
It will probably allow to remove `PluginClassLoaderRootPackageCustomizer` from `IndyBootstrap` and have only one such as well.